lib/cocoapods/resolver.rb

```diff
@@ -133,15 +150,10 @@ def resolver_specs_by_target
     #
     def search_for(dependency)
       @search[dependency] ||= begin
-        podfile_requirements = @podfile_requirements_by_root_name[dependency.root_name]
         additional_requirements = if locked_requirement = requirement_for_locked_pod_named(dependency.name)
-                                    if podfile_requirements
-                                      podfile_requirements + [locked_requirement]
-                                    else
-                                      [locked_requirement]
-                                    end
+                                    [locked_requirement]
                                   else
-                                    Array(podfile_requirements)
+                                    Array(@podfile_requirements_by_root_name[dependency.root_name])
```

Comment:
this differs from the previous behavior - is this correct? before we were including both podfile_requirements _and_ locked requirements if both were present

Reply:
Yes, see the commit message for https://github.com/CocoaPods/CocoaPods/pull/8708/commits/44f3054e0ce37e7c38ce2a8676c51f764f439259:

> Since the locked requirement will always be more constraining than the other requirements in the podfile

Reply:
ah I missed that - thanks for the details